app/src/main/java/campus/tech/kakao/map/MainViewModel.kt

```diff
@@ -33,11 +33,10 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
 
     fun addSearch(search: String) {
         val currentSearches = _savedSearches.value?.toMutableList() ?: mutableListOf()
-        if (!currentSearches.contains(search)) {
-            currentSearches.add(0, search)
-            _savedSearches.postValue(currentSearches)
-            saveSearchesToPreferences(currentSearches)
-        }
+        currentSearches.remove(search)
```

Comment:
nit; 아래처럼 구현도 가능합니다. :) 
```suggestion
        val old = _savedSearches.value ?: emptyList()
        val new = listOf(search) + (old - setOf(search))
```